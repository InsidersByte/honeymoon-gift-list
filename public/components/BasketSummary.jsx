/* @flow */

import React from 'react';
import { RaisedButton } from 'material-ui';
import { Link } from 'react-router';
import BasketSummaryTable from './BasketSummaryTable';
import { GIVER_ROUTE, HOME_ROUTE } from '../constants/routes';
import css from './BasketSummary.styl';

type PropsType = {
    basket: Map<number, Object>,
    basketCount: number,
    basketTotal: number,
    addToBasket: Function,
    removeFromBasket: Function,
    deleteFromBasket: Function,
};

const styles = {
    button: {
        marginTop: 12,
        marginRight: 12,
    },
};

export default function BasketSummary({ basket, basketCount, basketTotal, addToBasket, removeFromBasket, deleteFromBasket }: PropsType) {
    if (basketCount <= 0) {
        return (
            <div className={css.root}>
                <div className={css.container}>
                    <h1 className={css.title}>
                        Your Basket is empty!
                    </h1>

                    <div className={css.actions}>
                        <RaisedButton
                            primary
                            label="Back to Home"
                            containerElement={<Link to={HOME_ROUTE}>Back to Home</Link>}
                            linkButton
                        />
                    </div>
                </div>
            </div>
        );
    }

    return (
        <div className={css.root}>
            <div className={css.container}>
                <h1 className={css.title}>
                    Subtotal ({basketCount} items): Â£{basketTotal}
                </h1>

                <div className={css.content}>
                    <BasketSummaryTable
                        basket={basket}
                        basketTotal={basketTotal}
                        addToBasket={addToBasket}
                        removeFromBasket={removeFromBasket}
                        deleteFromBasket={deleteFromBasket}
                    />
                </div>

                <div className={css.actions}>
                    <RaisedButton
                        primary
                        label="Proceed to Checkout"
                        containerElement={<Link to={GIVER_ROUTE}>Proceed to Checkout</Link>}
                        linkButton
                        style={styles.button}
                    />

                    <RaisedButton
                        label="Back to Home"
                        containerElement={<Link to={HOME_ROUTE}>Back to Home</Link>}
                        linkButton
                        style={styles.button}
                    />
                </div>
            </div>
        </div>
    );
}
