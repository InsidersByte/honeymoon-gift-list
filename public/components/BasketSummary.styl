.root
    align-items center
    display flex
    flex-direction column
    justify-content center
    margin 0 10px
    min-height 75%

.container
    max-width 600px
    min-width 50%

.title
    font-size 2em
    margin-bottom 30px
    text-align center

.content
    margin-bottom 20px

.actions
    text-align center

    *:not(:first-child)
        margin-left 5px

@media (max-width 736px)
    .root
        font-size smaller
