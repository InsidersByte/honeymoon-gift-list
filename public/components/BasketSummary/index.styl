.root
    align-items center
    display flex
    flex-direction column
    justify-content center
    margin 15px 10px
    min-height 75%

.container
    min-width 25%

.title
    font-size 2em
    margin-bottom 30px
    text-align center

.content
    margin-bottom 20px

.actions
    text-align center

@media (max-width 736px)
    .root
        font-size smaller
