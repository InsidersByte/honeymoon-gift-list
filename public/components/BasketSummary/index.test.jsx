import React from 'react';
import { shallow } from 'enzyme';
import BasketSummary from './';

const item1 = { id: 1, price: 1, quantity: 1, remaining: 10 };
const item2 = { id: 2, price: 2, quantity: 2, remaining: 20 };

const basket = new Map();
basket.set(1, item1);
basket.set(2, item2);

const addToBasket = jest.fn();
const removeFromBasket = jest.fn();
const deleteFromBasket = jest.fn();

const props = {
    basket,
    basketCount: 3,
    basketTotal: 5,
    addToBasket,
    removeFromBasket,
    deleteFromBasket,
};

describe('BasketSummary', () => {
    afterEach(() => {
        expect(addToBasket).not.toHaveBeenCalled();
        expect(removeFromBasket).not.toHaveBeenCalled();
        expect(deleteFromBasket).not.toHaveBeenCalled();
    });

    it('should render correctly', () => {
        const wrapper = shallow(
            <BasketSummary {...props} />,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should singularise items if there is only one item in the basket', () => {
        const wrapper = shallow(
            <BasketSummary {...props} basketCount={1} />,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should render if the basket count is 0', () => {
        const wrapper = shallow(
            <BasketSummary {...props} basketCount={0} />,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should render <BasketSummaryListItem /> components', () => {
        const wrapper = shallow(
            <BasketSummary {...props} />,
        );

        expect(wrapper.find('BasketSummaryListItem').length).toBe(2);
    });

    it('should set <BasketSummaryListItem /> props', () => {
        const wrapper = shallow(
            <BasketSummary {...props} />,
        );

        const basketSummary = wrapper.find('BasketSummaryListItem');

        expect(basketSummary.at(0).prop('item')).toBe(item1);
        expect(basketSummary.at(0).prop('addToBasket')).toBe(addToBasket);
        expect(basketSummary.at(0).prop('removeFromBasket')).toBe(removeFromBasket);
        expect(basketSummary.at(0).prop('deleteFromBasket')).toBe(deleteFromBasket);

        expect(basketSummary.at(1).prop('item')).toBe(item2);
        expect(basketSummary.at(1).prop('addToBasket')).toBe(addToBasket);
        expect(basketSummary.at(1).prop('removeFromBasket')).toBe(removeFromBasket);
        expect(basketSummary.at(1).prop('deleteFromBasket')).toBe(deleteFromBasket);
    });
});
