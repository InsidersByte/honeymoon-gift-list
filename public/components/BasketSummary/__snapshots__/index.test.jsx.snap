// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasketSummary should render correctly 1`] = `
<div
  className="root"
>
  <div
    className="container"
  >
    <h1
      className="title"
    >
      Subtotal (
      3
       item
      s
      ): £
      5
    </h1>
    <Divider
      inset={false}
    />
    <div
      className="content"
    >
      <div>
        <BasketSummaryListItem
          addToBasket={[Function]}
          deleteFromBasket={[Function]}
          item={
            Object {
              "id": 1,
              "price": 1,
              "quantity": 1,
              "remaining": 10,
            }
          }
          removeFromBasket={[Function]}
        />
        <Divider
          inset={false}
        />
      </div>
      <div>
        <BasketSummaryListItem
          addToBasket={[Function]}
          deleteFromBasket={[Function]}
          item={
            Object {
              "id": 2,
              "price": 2,
              "quantity": 2,
              "remaining": 20,
            }
          }
          removeFromBasket={[Function]}
        />
        <Divider
          inset={false}
        />
      </div>
    </div>
    <div
      className="actions"
    >
      <RaisedButton
        containerElement={
          <Link
            replace={false}
            to="giver"
          >
            Proceed to Checkout
          </Link>
        }
        disabled={false}
        fullWidth={false}
        label="Proceed to Checkout"
        labelPosition="after"
        linkButton={true}
        primary={true}
        secondary={false}
        style={
          Object {
            "marginRight": 12,
            "marginTop": 12,
          }
        }
      />
      <RaisedButton
        containerElement={
          <Link
            replace={false}
            to="/"
          >
            Back to Home
          </Link>
        }
        disabled={false}
        fullWidth={false}
        label="Back to Home"
        labelPosition="after"
        linkButton={true}
        primary={false}
        secondary={false}
        style={
          Object {
            "marginRight": 12,
            "marginTop": 12,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`BasketSummary should render if the basket count is 0 1`] = `
<div
  className="root"
>
  <div
    className="container"
  >
    <h1
      className="title"
    >
      Your Basket is empty!
    </h1>
    <div
      className="actions"
    >
      <RaisedButton
        containerElement={
          <Link
            replace={false}
            to="/"
          >
            Back to Home
          </Link>
        }
        disabled={false}
        fullWidth={false}
        label="Back to Home"
        labelPosition="after"
        linkButton={true}
        primary={true}
        secondary={false}
      />
    </div>
  </div>
</div>
`;

exports[`BasketSummary should singularise items if there is only one item in the basket 1`] = `
<div
  className="root"
>
  <div
    className="container"
  >
    <h1
      className="title"
    >
      Subtotal (
      1
       item
      ): £
      5
    </h1>
    <Divider
      inset={false}
    />
    <div
      className="content"
    >
      <div>
        <BasketSummaryListItem
          addToBasket={[Function]}
          deleteFromBasket={[Function]}
          item={
            Object {
              "id": 1,
              "price": 1,
              "quantity": 1,
              "remaining": 10,
            }
          }
          removeFromBasket={[Function]}
        />
        <Divider
          inset={false}
        />
      </div>
      <div>
        <BasketSummaryListItem
          addToBasket={[Function]}
          deleteFromBasket={[Function]}
          item={
            Object {
              "id": 2,
              "price": 2,
              "quantity": 2,
              "remaining": 20,
            }
          }
          removeFromBasket={[Function]}
        />
        <Divider
          inset={false}
        />
      </div>
    </div>
    <div
      className="actions"
    >
      <RaisedButton
        containerElement={
          <Link
            replace={false}
            to="giver"
          >
            Proceed to Checkout
          </Link>
        }
        disabled={false}
        fullWidth={false}
        label="Proceed to Checkout"
        labelPosition="after"
        linkButton={true}
        primary={true}
        secondary={false}
        style={
          Object {
            "marginRight": 12,
            "marginTop": 12,
          }
        }
      />
      <RaisedButton
        containerElement={
          <Link
            replace={false}
            to="/"
          >
            Back to Home
          </Link>
        }
        disabled={false}
        fullWidth={false}
        label="Back to Home"
        labelPosition="after"
        linkButton={true}
        primary={false}
        secondary={false}
        style={
          Object {
            "marginRight": 12,
            "marginTop": 12,
          }
        }
      />
    </div>
  </div>
</div>
`;
